chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -115,76 +150,107 @@ public void updateChebiReferenceMolecules() throws SQLException, Exception
 			String chebiName = entity.getChebiAsciiName();
 			List<DataItem> chebiFormulae = entity.getFormulae();
 
-			updateReferenceEntities(molecule, chebiName, instanceEdit);
 
-			// Now, check to see if we need to update the ReferenceMolecule itself.
 			String moleculeIdentifier = (String) molecule.getAttributeValue(ReactomeJavaConstants.identifier);
 			String moleculeName = (String) molecule.getAttributeValuesList(ReactomeJavaConstants.name).get(0);
 			String moleculeFormulae = (String) molecule.getAttributeValue(ReactomeJavaConstants.formula);
 
 			String prefix = "ReferenceMolecule (DB ID: " + molecule.getDBID() + " / ChEBI ID: " + moleculeIdentifier + ") has changes: ";
 
-			updateMoleculeIdentifier(molecule, chebiID, moleculeIdentifier, prefix);
-			updateMoleculeName(molecule, chebiName, moleculeName, prefix);
-			updateMoleculeFormula(molecule, chebiFormulae, moleculeFormulae, prefix);
-			// Update the display name.
-			InstanceDisplayNameGenerator.setDisplayName(molecule);
-			adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants._displayName);
-			
-			if (!testMode)
+			reportIfMoleculeIdentifierChanged(molecule, chebiID, moleculeIdentifier, prefix);
+			boolean nameUpdated = updateMoleculeName(molecule, chebiName, moleculeName, prefix);
+			boolean formulaUpdated = updateMoleculeFormula(molecule, chebiFormulae, moleculeFormulae, prefix);
+			// If the ChEBI name was updated, update the referenceEntities that refer to the molecule.
+			if (nameUpdated)
+			{
+				updateReferenceEntities(molecule, chebiName, instanceEdit);
+			}
+			// Now, check to see if we need to update the ReferenceMolecule itself.
+			if (nameUpdated || formulaUpdated)
 			{
 				addInstanceEditToExistingModifieds(instanceEdit, molecule);
-				
-				if (adaptor.supportsTransactions())
-				{
-					adaptor.commit();
-				}
+				// Update the display name.
+				InstanceDisplayNameGenerator.setDisplayName(molecule);
+				adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants._displayName);
 			}
 		}
+		if (!testMode)
+		{
+			adaptor.commit();
+		}
+		else
+		{
+			adaptor.rollback();
+		}
+		
```

Comment:
Might be more succinct as ternary - !testmode ? adaptor.commit() : adaptor.rollback();

Reply:
That would be more succinct, but I prefer the clarity of what is written. I expect a ternary operator to be used in an expression that returns, such as:

```
int x = someNumber > 0 ? processPositive(someNumber) : processNegative(someNumber);
```

A ternary that does not produce a value is odd-looking to me. I don't think this is commonly done in Java, and I don't recall seeing this practice.

Reply:
UPDATE: I just tried this and Java does not seem to allow this:
```
0 < 1 ? System.exit(0) : System.exit(1);
```
Error: "The left-hand side of an assignment must be a variable"

Reply:
...I guess that's why I've never seen it.